libp2p/stream.go

```diff
@@ -0,0 +1,62 @@
+package libp2p
+
+import (
+	"bytes"
+	"io"
+	"time"
+
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/rlp"
+	net "github.com/libp2p/go-libp2p-net"
+)
+
+func NewStream(s net.Stream, r, w time.Duration) Stream {
+	return Stream{
+		s:            s,
+		rlp:          rlp.NewStream(s, 0),
+		readTimeout:  r,
+		writeTimeout: w,
+	}
+}
+
+type Stream struct {
+	readTimeout, writeTimeout time.Duration
+
+	rlp *rlp.Stream
+	s   net.Stream
+}
+
+func (s Stream) ReadMsg() (msg p2p.Msg, err error) {
+	if s.readTimeout != 0 {
+		if err = s.s.SetReadDeadline(time.Now().Add(s.readTimeout)); err != nil {
```

Comment:
`s.s` is meaningless in this context. I am not sure what is better, though. The only thing I figured out is `s.originalStream` because `s` is a libp2p to rlp transform stream and provided libp2p stream is "original".